src/openms/source/FILTERING/DATAREDUCTION/Deisotoper.cpp

```diff
@@ -113,7 +118,7 @@ void Deisotoper::deisotopeAndSingleCharge(MSSpectrum& spec,
           {
             // Possible improvement: include proper averagine model filtering. for now start at the second peak to test hypothesis
             // Note: this is a common approach used in several other search engines
-            if (old_spectrum[p].getIntensity() > old_spectrum[extensions.back()].getIntensity())
+            if (model_decreasing && old_spectrum[p].getIntensity() > old_spectrum[extensions.back()].getIntensity())
```

Comment:
```suggestion
            if (model_decreasing && extensions.size > 2 && old_spectrum[p].getIntensity() > old_spectrum[extensions.back()].getIntensity())
```

@timosachsenberg is that what you had in mind?